From 1287f3bb5dfe4cfbae470b54ccb341cd1cee82c5 Mon Sep 17 00:00:00 2001
From: Jake <jake@fulu.org>
Date: Thu, 9 Jul 2026 19:53:26 +0100
Subject: [PATCH] Fix setPageTitle() Message-argument deprecation warnings

Passing a Message object straight to OutputPage::setPageTitle() was
deprecated in MediaWiki 1.41

---
 .../includes/specials/SpecialRemoveMasterSystemGift.php     | 6 +++---
 SystemGifts/includes/specials/SpecialViewSystemGift.php     | 6 +++---
 SystemGifts/includes/specials/SpecialViewSystemGifts.php    | 6 +++---
 SystemGifts/includes/specials/TopAwards.php                 | 2 +-
 UserProfile/includes/specials/SpecialEditProfile.php        | 2 +-
 5 files changed, 11 insertions(+), 11 deletions(-)

diff --git a/SystemGifts/includes/specials/SpecialRemoveMasterSystemGift.php b/SystemGifts/includes/specials/SpecialRemoveMasterSystemGift.php
index 02d39647..c2d4e26f 100644
--- a/SystemGifts/includes/specials/SpecialRemoveMasterSystemGift.php
+++ b/SystemGifts/includes/specials/SpecialRemoveMasterSystemGift.php
@@ -78,7 +78,7 @@ class RemoveMasterSystemGift extends UnlistedSpecialPage {
 		$this->gift_id = $request->getInt( 'gift_id', $par );
 
 		if ( !$this->gift_id ) {
-			$out->setPageTitle( $this->msg( 'ga-error-title' ) );
+			$out->setPageTitle( $this->msg( 'ga-error-title' )->text() );
 			$out->addHTML( $this->msg( 'ga-error-message-invalid-link' )->escaped() );
 			return;
 		}
@@ -109,7 +109,7 @@ class RemoveMasterSystemGift extends UnlistedSpecialPage {
 			$this->deleteImage( $this->gift_id, 'l' );
 			$this->deleteImage( $this->gift_id, 'ml' );
 
-			$out->setPageTitle( $this->msg( 'ga-remove-success-title', $gift['gift_name'] ) );
+			$out->setPageTitle( $this->msg( 'ga-remove-success-title', $gift['gift_name'] )->text() );
 
 			$output = '<div class="back-links">
 				<a href="' . htmlspecialchars( SpecialPage::getTitleFor( 'SystemGiftManager' )->getFullURL() ) . '">' .
@@ -138,7 +138,7 @@ class RemoveMasterSystemGift extends UnlistedSpecialPage {
 		$systemGiftIcon = new SystemGiftIcon( $this->gift_id, 'l' );
 		$icon = $systemGiftIcon->getIconHTML();
 
-		$this->getOutput()->setPageTitle( $this->msg( 'ga-remove-title', $gift['gift_name'] ) );
+		$this->getOutput()->setPageTitle( $this->msg( 'ga-remove-title', $gift['gift_name'] )->text() );
 
 		$output = '<div class="back-links">
 			<a href="' . htmlspecialchars( SpecialPage::getTitleFor( 'SystemGiftManager' )->getFullURL() ) . '">' .
diff --git a/SystemGifts/includes/specials/SpecialViewSystemGift.php b/SystemGifts/includes/specials/SpecialViewSystemGift.php
index d5f8c146..7a10be3d 100644
--- a/SystemGifts/includes/specials/SpecialViewSystemGift.php
+++ b/SystemGifts/includes/specials/SpecialViewSystemGift.php
@@ -39,7 +39,7 @@ class ViewSystemGift extends UnlistedSpecialPage {
 		// numeric, display an error message
 		$giftId = $this->getRequest()->getInt( 'gift_id' );
 		if ( !$giftId ) {
-			$out->setPageTitle( $this->msg( 'ga-error-title' ) );
+			$out->setPageTitle( $this->msg( 'ga-error-title' )->text() );
 			$out->addHTML( $this->msg( 'ga-error-message-invalid-link' )->escaped() );
 			return;
 		}
@@ -76,7 +76,7 @@ class ViewSystemGift extends UnlistedSpecialPage {
 				[ 'actor' => [ 'JOIN', 'actor_id = sg_actor' ] ]
 			);
 
-			$out->setPageTitle( $this->msg( 'ga-gift-title', $gift['user_name'], $gift['name'] ) );
+			$out->setPageTitle( $this->msg( 'ga-gift-title', $gift['user_name'], $gift['name'] )->text() );
 
 			$output .= '<div class="back-links">' .
 				$this->msg( 'ga-back-link', $gift['user_name'] )->parse() .
@@ -133,7 +133,7 @@ class ViewSystemGift extends UnlistedSpecialPage {
 
 			$out->addHTML( $output );
 		} else {
-			$out->setPageTitle( $this->msg( 'ga-error-title' ) );
+			$out->setPageTitle( $this->msg( 'ga-error-title' )->text() );
 			$out->addHTML( $this->msg( 'ga-error-message-invalid-link' )->escaped() );
 		}
 	}
diff --git a/SystemGifts/includes/specials/SpecialViewSystemGifts.php b/SystemGifts/includes/specials/SpecialViewSystemGifts.php
index 5921e65d..7c5d411d 100644
--- a/SystemGifts/includes/specials/SpecialViewSystemGifts.php
+++ b/SystemGifts/includes/specials/SpecialViewSystemGifts.php
@@ -51,7 +51,7 @@ class ViewSystemGifts extends SpecialPage {
 		 * It will automatically return them to the ViewSystemGifts page
 		 */
 		if ( !$currentUser->isRegistered() && $user_name == '' ) {
-			$out->setPageTitle( $this->msg( 'ga-error-title' ) );
+			$out->setPageTitle( $this->msg( 'ga-error-title' )->text() );
 			$login = SpecialPage::getTitleFor( 'Userlogin' );
 			$out->redirect( htmlspecialchars( $login->getFullURL( 'returnto=Special:ViewSystemGifts' ) ) );
 			return;
@@ -69,7 +69,7 @@ class ViewSystemGifts extends SpecialPage {
 		 * Error message for username that does not exist (from URL)
 		 */
 		if ( $targetUser->getId() == 0 ) {
-			$out->setPageTitle( $this->msg( 'ga-error-title' ) );
+			$out->setPageTitle( $this->msg( 'ga-error-title' )->text() );
 			$out->addHTML( $this->msg( 'ga-error-message-no-user' )->escaped() );
 			return;
 		}
@@ -92,7 +92,7 @@ class ViewSystemGifts extends SpecialPage {
 		/**
 		 * Show gift count for user
 		 */
-		$out->setPageTitle( $this->msg( 'ga-title', $rel->user_name ) );
+		$out->setPageTitle( $this->msg( 'ga-title', $rel->user_name )->text() );
 
 		$output .= '<div class="back-links">' .
 			$this->msg(
diff --git a/SystemGifts/includes/specials/TopAwards.php b/SystemGifts/includes/specials/TopAwards.php
index 141614b0..b1e8f2ef 100644
--- a/SystemGifts/includes/specials/TopAwards.php
+++ b/SystemGifts/includes/specials/TopAwards.php
@@ -125,7 +125,7 @@ class TopAwards extends UnlistedSpecialPage {
 		// topawards-comment-title, topawards-recruit-title,
 		// topawards-friend-title
 		$out->setPageTitle(
-			$this->msg( 'topawards-' . strtolower( $page_category ) . '-title' )
+			$this->msg( 'topawards-' . strtolower( $page_category ) . '-title' )->text()
 		);
 
 		// Add CSS
diff --git a/UserProfile/includes/specials/SpecialEditProfile.php b/UserProfile/includes/specials/SpecialEditProfile.php
index 0485cd88..9036a570 100644
--- a/UserProfile/includes/specials/SpecialEditProfile.php
+++ b/UserProfile/includes/specials/SpecialEditProfile.php
@@ -146,7 +146,7 @@ class SpecialEditProfile extends SpecialUpdateProfile {
 		// Needed because display*Form() functions can and do override our title so
 		// if we don't do this here, the page title ends up being something like
 		// "Other information" and the HTML title ends up being "Tidbits"
-		$out->setPageTitle( $this->msg( 'editprofile' ) );
+		$out->setPageTitle( $this->msg( 'editprofile' )->text() );
 		$out->setHTMLTitle( $this->msg( 'pagetitle', $this->msg( 'edit-profiles-title' ) ) );
 	}
 
